libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -962,6 +965,14 @@ private async Task<MicrosoftAppCredentials> GetAppCredentialsAsync(string appId,
                 return appCredentials;
             }
 
+            var provider2 = _credentialProvider as ICredentialProvider2;
```

Comment:
Please no number in variables or names. consider a meaningful name that for example hints the purpose of the provider